### PR TITLE
Use a helper to find built binaries

### PR DESCRIPTION
**Pull request by eparis**
We should not be determining the arch/os and all the possible
directories binaries can live. Let a hack/lib/ helper do it for us.

### CODE REVIEW

File: cluster/kubectl.sh [hunk 1]
```diff
@@ -20,6 +20,7 @@ set -o pipefail
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/cluster/kube-env.sh"
+source "${kUBE_ROOT}/hack/lib/util.sh"
```
Comment (dchen1107) [inline cluster/kubectl.sh]: s/kUBE_ROOT/KUBE_ROOT?

Comment (eparis) [inline cluster/kubectl.sh]: whoops!

**Comment by eparis** [comment]
fixed

File: hack/ginkgo-e2e.sh [hunk 2]
```diff
@@ -75,6 +32,7 @@ e2e=$( (ls -t "${locations[@]}" 2>/dev/null || true) | head -1 )
 export KUBECTL KUBE_CONFIG_FILE
 
 source "${KUBE_ROOT}/cluster/kube-env.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
```
Comment (eparis) [inline hack/ginkgo-e2e.sh]: now I'm not so sure, did I just source after I used it...

**Comment by dchen1107** [comment]
LGTM. Will merge on green.

**Comment by justinsb** [comment]
This seems to have broken e2e tests, for me.

It looks like hack/lib isn't copied into a staging area of some sort.  Trying to track this down.

**Comment by eparis** [comment]
want me to revert while you run it down?

**Comment by ixdy** [comment]
Yes, this also broke Jenkins.

/jenkins-master-data/jobs/kubernetes-e2e-gce/workspace/kubernetes/hack/e2e-internal/../../cluster/../cluster/gce/../../cluster/../cluster/kubectl.sh: line 23: /jenkins-master-data/jobs/kubernetes-e2e-gce/workspace/kubernetes/hack/e2e-internal/../../cluster/../cluster/gce/../../cluster/../cluster/../hack/lib/util.sh: No such file or directory
2015/04/14 09:36:37 Error running up: exit status 1
2015/04/14 09:36:37 Error starting e2e cluster. Aborting.
exit status 1

**Comment by quinton-hoole** [comment]
eparis@, dchan1107@ : please at least run the e2e tests once before merging in future.  We're working on having that wired into Github properly, but in the mean time please run them locally.

**Comment by eparis** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/6815

**Comment by dchen1107** [comment]
Yes, I noticed Jerkins' breakage, and chatted with @quinton-hoole, thought it was due to disk issue. Sorry for the breakage.

**Pull request merged**